src/Core/src/Response.php

```diff
@@ -140,14 +245,20 @@ public function getContent(): string
     {
         $this->resolve();
 
-        return $this->httpResponse->getContent(false);
+        $content = $this->httpResponse->getContent(false);
+        $this->bodyDownloaded = true;
+
+        return $content;
```

Comment:
This is a trick @nicolas-grekas has showed me =)

```suggestion
        try {
            return $this->httpResponse->getContent(false);
        } finally {
            $this->bodyDownloaded = true;
        }
```

Reply:
not equivalent in case an error happens, though.

Reply:
ahahah.. I love that syntax, but thought you didn't like it.. 
happy to apply!